api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.db.ConditionDAO;
+import org.slf4j.Logger;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods for the condition domain object
+ *
+ * @since 2.2
+ */
+@Transactional
+public class ConditionServiceImpl extends BaseOpenmrsService implements ConditionService{
+
+	private static final Logger log = org.slf4j.LoggerFactory.getLogger(ConditionServiceImpl.class);
+
+	private ConditionDAO conditionDAO;
+
+	private ConditionService conditionService;
+	
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	public Condition save(Condition condition) {
+		if(condition.getCondition() == null){
+			throw new APIException("The condition parameter is required");
+		}
+		if(condition.getClinicalStatus() == null){
+			throw new APIException("The clinical status is required");
+		}
+		Condition existingCondition = getConditionByUuid(condition.getUuid());
+		if(condition.equals(existingCondition)){
+			return existingCondition;
+		}
+		if(existingCondition == null){
+			return conditionDAO.saveCondition(condition);
+		}
+		if (log.isInfoEnabled()) {
+			log.info("Saving condition " + condition);
+		}
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Voids a condition
+	 *
+	 * @param condition  - the condition to be voided
+	 * @param voidReason - the reason for voiding the condition
+	 */
+	@Override
+	public Condition voidCondition(Condition condition, String voidReason) {
+		if(voidReason == null || !StringUtils.hasLength(voidReason)){
+			throw new IllegalArgumentException("The 'voidReason' parameter is required to void this condition");
+		}
+		if(log.isInfoEnabled()){
+			log.info("Voiding condition " + condition);
+		}
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Gets a condition based on the uuid
+	 *
+	 * @param uuid - uuid of the condition to be returned
+	 * @return the condition
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getConditionByUuid(String uuid) {
+		if(log.isInfoEnabled()){
+			log.info("Getting condition with uuid " + uuid);
+		}
+		return conditionDAO.getConditionByUuid(uuid);
+	}
+
+	/**
+	 * Gets a patient's active conditions
+	 *
+	 * @param patient - the patient to retrieve conditions for
+	 * @return a list of the patient's active conditions
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getActiveConditions(Patient patient) {
+		if(log.isInfoEnabled()){
+			log.info("Getting active conditions with patient " + patient);
+		}
+		return conditionDAO.getActiveConditions(patient);
+	}
+
+	public void setConditionService(ConditionService conditionService) {
+		this.conditionService = conditionService;
```

Comment:
What are you using this for?

Reply:
They are not necessary. I have modified accordingly.